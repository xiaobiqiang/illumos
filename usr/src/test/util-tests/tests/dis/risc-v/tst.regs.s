/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2018, Joyent, Inc.
 */

/*
 * Test that we properly name all registers in their place.
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	add	x0, x0, x0
	add	ra, ra, ra
	add	sp, sp, sp
	add	gp, gp, gp
	add	tp, tp, tp
	add	t0, t0, t0
	add	t1, t1, t1
	add	t2, t2, t2
	add	s0, s0, s0
	add	s1, s1, s1
	add	a0, a0, a0
	add	a1, a1, a1
	add	a2, a2, a2
	add	a3, a3, a3
	add	a4, a4, a4
	add	a5, a5, a5
	add	a6, a6, a6
	add	a7, a7, a7
	add	s2, s2, s2
	add	s3, s3, s3
	add	s4, s4, s4
	add	s5, s5, s5
	add	s6, s6, s6
	add	s7, s7, s7
	add	s8, s8, s8
	add	s9, s9, s9
	add	s10, s10, s10
	add	s11, s11, s11
	add	t3, t3, t3
	add	t4, t4, t4
	add	t5, t5, t5
	add	t6, t6, t6

	add	x0, ra, sp
	add	ra, sp, gp
	add	sp, gp, tp
	add	gp, tp, t0
	add	tp, t0, t1
	add	t0, t1, t2
	add	t1, t2, s0
	add	t2, s0, s1
	add	s0, s1, a0
	add	s1, a0, a1
	add	a0, a1, a2
	add	a1, a2, a3
	add	a2, a3, a4
	add	a3, a4, a5
	add	a4, a5, a6
	add	a5, a6, a7
	add	a6, a7, s2
	add	a7, s2, s3
	add	s2, s3, s4
	add	s3, s4, s5
	add	s4, s5, s6
	add	s5, s6, s7
	add	s6, s7, s8
	add	s7, s8, s9
	add	s8, s9, s10
	add	s9, s10, s11
	add	s10, s11, t3
	add	s11, t3, t4
	add	t3, t4, t5
	add	t4, t5, t6
	add	t5, t6, x0
	add	t6, x0, ra
.size libdis_test, [.-libdis_test]
