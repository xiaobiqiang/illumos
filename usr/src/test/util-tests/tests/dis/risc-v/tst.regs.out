    libdis_test:       33 00 00 00        add x0,x0,x0
    libdis_test+0x4:   b3 80 10 00        add ra,ra,ra
    libdis_test+0x8:   33 01 21 00        add sp,sp,sp
    libdis_test+0xc:   b3 81 31 00        add gp,gp,gp
    libdis_test+0x10:  33 02 42 00        add tp,tp,tp
    libdis_test+0x14:  b3 82 52 00        add t0,t0,t0
    libdis_test+0x18:  33 03 63 00        add t1,t1,t1
    libdis_test+0x1c:  b3 83 73 00        add t2,t2,t2
    libdis_test+0x20:  33 04 84 00        add s0,s0,s0
    libdis_test+0x24:  b3 84 94 00        add s1,s1,s1
    libdis_test+0x28:  33 05 a5 00        add a0,a0,a0
    libdis_test+0x2c:  b3 85 b5 00        add a1,a1,a1
    libdis_test+0x30:  33 06 c6 00        add a2,a2,a2
    libdis_test+0x34:  b3 86 d6 00        add a3,a3,a3
    libdis_test+0x38:  33 07 e7 00        add a4,a4,a4
    libdis_test+0x3c:  b3 87 f7 00        add a5,a5,a5
    libdis_test+0x40:  33 08 08 01        add a6,a6,a6
    libdis_test+0x44:  b3 88 18 01        add a7,a7,a7
    libdis_test+0x48:  33 09 29 01        add s2,s2,s2
    libdis_test+0x4c:  b3 89 39 01        add s3,s3,s3
    libdis_test+0x50:  33 0a 4a 01        add s4,s4,s4
    libdis_test+0x54:  b3 8a 5a 01        add s5,s5,s5
    libdis_test+0x58:  33 0b 6b 01        add s6,s6,s6
    libdis_test+0x5c:  b3 8b 7b 01        add s7,s7,s7
    libdis_test+0x60:  33 0c 8c 01        add s8,s8,s8
    libdis_test+0x64:  b3 8c 9c 01        add s9,s9,s9
    libdis_test+0x68:  33 0d ad 01        add s10,s10,s10
    libdis_test+0x6c:  b3 8d bd 01        add s11,s11,s11
    libdis_test+0x70:  33 0e ce 01        add t3,t3,t3
    libdis_test+0x74:  b3 8e de 01        add t4,t4,t4
    libdis_test+0x78:  33 0f ef 01        add t5,t5,t5
    libdis_test+0x7c:  b3 8f ff 01        add t6,t6,t6
    libdis_test+0x80:  33 80 20 00        add x0,ra,sp
    libdis_test+0x84:  b3 00 31 00        add ra,sp,gp
    libdis_test+0x88:  33 81 41 00        add sp,gp,tp
    libdis_test+0x8c:  b3 01 52 00        add gp,tp,t0
    libdis_test+0x90:  33 82 62 00        add tp,t0,t1
    libdis_test+0x94:  b3 02 73 00        add t0,t1,t2
    libdis_test+0x98:  33 83 83 00        add t1,t2,s0
    libdis_test+0x9c:  b3 03 94 00        add t2,s0,s1
    libdis_test+0xa0:  33 84 a4 00        add s0,s1,a0
    libdis_test+0xa4:  b3 04 b5 00        add s1,a0,a1
    libdis_test+0xa8:  33 85 c5 00        add a0,a1,a2
    libdis_test+0xac:  b3 05 d6 00        add a1,a2,a3
    libdis_test+0xb0:  33 86 e6 00        add a2,a3,a4
    libdis_test+0xb4:  b3 06 f7 00        add a3,a4,a5
    libdis_test+0xb8:  33 87 07 01        add a4,a5,a6
    libdis_test+0xbc:  b3 07 18 01        add a5,a6,a7
    libdis_test+0xc0:  33 88 28 01        add a6,a7,s2
    libdis_test+0xc4:  b3 08 39 01        add a7,s2,s3
    libdis_test+0xc8:  33 89 49 01        add s2,s3,s4
    libdis_test+0xcc:  b3 09 5a 01        add s3,s4,s5
    libdis_test+0xd0:  33 8a 6a 01        add s4,s5,s6
    libdis_test+0xd4:  b3 0a 7b 01        add s5,s6,s7
    libdis_test+0xd8:  33 8b 8b 01        add s6,s7,s8
    libdis_test+0xdc:  b3 0b 9c 01        add s7,s8,s9
    libdis_test+0xe0:  33 8c ac 01        add s8,s9,s10
    libdis_test+0xe4:  b3 0c bd 01        add s9,s10,s11
    libdis_test+0xe8:  33 8d cd 01        add s10,s11,t3
    libdis_test+0xec:  b3 0d de 01        add s11,t3,t4
    libdis_test+0xf0:  33 8e ee 01        add t3,t4,t5
    libdis_test+0xf4:  b3 0e ff 01        add t4,t5,t6
    libdis_test+0xf8:  33 8f 0f 00        add t5,t6,x0
    libdis_test+0xfc:  b3 0f 10 00        add t6,x0,ra
